pkg/cli/zip_test.go

```diff
@@ -838,3 +846,123 @@ func TestNodeRangeSelection(t *testing.T) {
 		}
 	}
 }
+
+// Check if export request is from a lease for a descriptor to avoid picking
+// up on wrong export requests
+func isLeasingExportRequest(r *kvpb.ExportRequest) bool {
+	_, tenantID, _ := keys.DecodeTenantPrefix(r.Key)
+	codec := keys.MakeSQLCodec(tenantID)
+	return bytes.HasPrefix(r.Key, codec.DescMetadataPrefix()) &&
+		r.EndKey.Equal(r.Key.PrefixEnd())
+}
+
+func TestZipJobTrace(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	var continueCh chan struct{}
+	s, sqlDB, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Insecure:          true,
+		DefaultTestTenant: base.TestControlsTenantsExplicitly,
+		Knobs: base.TestingKnobs{
+			JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+			Store: &kvserver.StoreTestingKnobs{
+				TestingRequestFilter: func(ctx context.Context, ba *kvpb.BatchRequest) *kvpb.Error {
+					for _, ru := range ba.Requests {
+						switch r := ru.GetInner().(type) {
+						case *kvpb.ExportRequest:
+							if !isLeasingExportRequest(r) && continueCh != nil {
```

Comment:
I feel like I've soured a little on the request filter tests -- they're tedious to debug if/when some other bit of the server starts using the same requests or something.

Would this test be simpler if we just used a 5 row IMPORT job from workload and then exported `testingKnobs.afterImport` in pkg/sql/importer?

Reply:
chatted offline, decided it was easier to inject a FakeResumer instead to get rid of the request filters.